### ISSUE TITLE
MenuBar: Documentation seems incorrect regarding functionality

### ISSUE THREAD

**Issue by loafingbread** [godotengine/godot-docs#9724]
**Your Godot version:** 4.2

**Issue description:** Documentation inaccurately describes MenuBar's interactions with MenuButton and PopupMenu

**URL to the documentation page (if already existing):** https://docs.godotengine.org/en/stable/classes/class_menubar.html

The documentation page says the following: "A horizontal menu bar that creates a [MenuButton](https://docs.godotengine.org/en/stable/classes/class_menubutton.html#class-menubutton) for each [PopupMenu](https://docs.godotengine.org/en/stable/classes/class_popupmenu.html#class-popupmenu) child. New items are created by adding [PopupMenu](https://docs.godotengine.org/en/stable/classes/class_popupmenu.html#class-popupmenu)s to this node".

I tried using the MenuBar for the first time and had some trouble. It turns out that the the actual behavior of MenuBar is the reverse of what is describe in the documentation. It actually creates a "PopupMenu for each MenuButton" and not a "MenuButton for each PopupMenu".

I feel it is a bit misleading, but please let me know if I am interpreting the documentation incorrectly.

TLDR:
1. MenuBar says it creates a MenuButton for each PopupMenu, but it actually creates a PopupMenu for each MenuButton. 
2. The above is confusing, can the documentation be updated if it is found to be true.
3. Could an example snippet on basic/common usage for (MenuBar, MenuButton, PopupMenu) be added to the documentation. I am happy to create a pull request if needed.

**Scene Tree during Local/Compile**
- We can see that the first MenuButton has no PopupMenu child, while the second one does.
- During runtime, we will see the actual behavior of MenuBar
<img width="416" alt="image" src="https://github.com/user-attachments/assets/2a4b24e1-6141-451f-a942-4e8c9723dc2e">

**Scene Tree during Runtime/Remote**
- We can see that a PopupMenu is created at runtime ("@PopupMenu") for each MenuButton, even if it already has one.
<img width="407" alt="image" src="https://github.com/user-attachments/assets/10cb7a65-e824-4653-8dd1-afcb8faf4ec8">

**Comment by AThousandShips** [comment]
It doesn't create a plain `MenuButton` but its own custom thing, but it *does* respond to `PopupMenu` children, not `MenuButton` children, as you can see here in the actual code:
https://github.com/godotengine/godot/blob/c73ac74c4a44ccfed158b3603f809a24590bac0a/scene/gui/menu_bar.cpp#L545-L571

Your screenshots are a bit strange though, they do not represent why are you adding a `HBoxContainer`? The `MenuButton`s should be *children* of the `MenuBar`

**Comment by AThousandShips** [comment]
Closing due to lack of response, the menu bar and the documentation works correctly when used as intended, and I do not believe any further clarification is needed, if you feel the fact that the `PopupMenu`s have to be *direct* children then we can add that clarification, but otherwise there's no issue here

**Comment by loafingbread** [comment]
@AThousandShips Thank you so much for the response! I'll try what you suggested! 

Thank you again, and sorry for the wild goose chase. I might have gotten a bit confused hahaha!
